tests/console/systemd_rpm_macros.pm

```diff
@@ -0,0 +1,60 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: systemd-rpm-macros test
+#          - call a list of macros to make sure they are available (list taken from manual testing report)
+#          - if Tumbleweed or SLE >= 12-sp4, also download sources and install multipath-tools to run some macros
+#
+# Maintainer: Michael Grifalconi <mgrifalconi@suse.com>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils;
+use registration qw(add_suseconnect_product);
+
+sub build_mt {
+    zypper_call("in rpmbuild");
+    zypper_call("si multipath-tools");
+    assert_script_run 'rpmbuild -bb /usr/src/packages/SPECS/multipath-tools.spec';
+}
+
+
+sub run {
+    #Preparation
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    # Call macros to make sure they are available
+    zypper_call("in systemd-rpm-macros");
+    assert_script_run 'wget --quiet ' . data_url('console/test_systemd_rpm_macros.sh');
+    assert_script_run 'wget --quiet ' . data_url('console/systemd_rpm_macros_list');
+    assert_script_run 'chmod +x test_systemd_rpm_macros.sh';
+    assert_script_run "./test_systemd_rpm_macros.sh", 900;
+
+    # Test build of multipath-tools on tw, or SLE >= 12.4
```

Comment:
Good job MIchael :) I have a small question here, why build of multipath-tools is not also tested on 12-SP2 and 12-SP3 (as this test is scheduled to run also on these two versions )?

Reply:
That's a good question, I missed to write this somewhere :)

The build, unfortunately did not work in these 2 versions due to conflicts on build dependencies. It would have required a downgrade of some packages but in such old versions, there is no `zypper --allow-downgrade` option. After the test it would have required a cleanup as well or having its own test suite to skip the cleanup.

Since the regression of multipath-tools happened on 15-sp1 I went for a simpler solution and I would propose to wait and check if we really need something on 12.2/3 before investing more time on it, in that case, maybe a completely different package can be built.

What do you think?

Reply:
Yes, I agree with your logic, it's better to keep it like this ^^